crates/driver/src/infra/api/routes/notify/mod.rs

```diff
@@ -0,0 +1,23 @@
+mod dto;
+
+pub use dto::NotifyError;
+
+use crate::infra::api::{Error, State};
+
+pub(in crate::infra::api) fn notify(router: axum::Router<State>) -> axum::Router<State> {
+    router.route("/notify", axum::routing::post(route))
+}
+
+async fn route(
+    state: axum::extract::State<State>,
+    req: axum::Json<dto::NotifyRequest>,
+) -> Result<hyper::StatusCode, (hyper::StatusCode, axum::Json<Error>)> {
+    let solver = &state.solver().name().0;
+    tracing::trace!(?req, ?solver, "received a notification");
```

Comment:
nit: probably ok to set it to `debug` as we don't start with a lot of notifications.